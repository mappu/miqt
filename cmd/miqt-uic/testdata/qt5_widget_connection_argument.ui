<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QSpinBox" name="first"/>
   </item>
   <item>
    <widget class="QSpinBox" name="second"/>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>first</sender>
   <signal>valueChanged(int)</signal>
   <receiver>second</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
