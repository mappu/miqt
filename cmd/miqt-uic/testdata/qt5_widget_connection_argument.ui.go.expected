// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	"github.com/mappu/miqt/qt"
)

type FormUi struct {
	Form             *qt.QWidget
	horizontalLayout *qt.QHBoxLayout
	first            *qt.QSpinBox
	second           *qt.QSpinBox
}

// NewFormUi creates all Qt widget classes for Form.
func NewFormUi() *FormUi {
	ui := &FormUi{}
	ui.Form = qt.NewQWidget(nil)
	ui.Form.SetObjectName("Form")
	ui.Form.Resize(400, 300)
	ui.horizontalLayout = qt.NewQHBoxLayout(ui.Form)
	ui.horizontalLayout.SetObjectName("horizontalLayout")
	ui.horizontalLayout.SetContentsMargins(11, 11, 11, 11)
	ui.horizontalLayout.SetSpacing(6)
	ui.first = qt.NewQSpinBox(ui.Form)
	ui.first.SetObjectName("first")

	ui.horizontalLayout.AddWidget(ui.first.QWidget)
	ui.second = qt.NewQSpinBox(ui.Form)
	ui.second.SetObjectName("second")

	ui.horizontalLayout.AddWidget(ui.second.QWidget)

	ui.first.OnValueChanged(ui.second.SetValue)

	ui.Retranslate()

	return ui
}

// Retranslate reapplies all text translations.
func (ui *FormUi) Retranslate() {
	ui.Form.SetWindowTitle(qt.QCoreApplication_Tr("Form"))
}
